code/game/objects/items/books/skill_book.dm

```diff
@@ -633,11 +642,143 @@ MEDICAL
+	var/list/progress_messages =list("Still quite a few blank pages left.","Feels like you're near halfway done.","You've made good progress.","Just needs a few finishing touches.","And then finish it. Done!") // Messages are in order of progress.
+	var/writing_time = 15 SECONDS // time it takes to write a segment of the book. This happens 6 times total
+
+//these all show up in the book fabricator
+/obj/item/book/skill/custom/black
+	icon_state = "book1"
+/obj/item/book/skill/custom/yellow
+	icon_state = "book3"
+/obj/item/book/skill/custom/blue
+	icon_state = "book4"
+/obj/item/book/skill/custom/green
+	icon_state = "book5"
+/obj/item/book/skill/custom/purple
+	icon_state = "book6"
+/obj/item/book/skill/custom/light_blue
+	icon_state = "book7"
+
+/obj/item/book/skill/custom/attackby(obj/item/W, mob/user)
+	if(istype(W, /obj/item/pen))
+
+		if(!user.skill_check(SKILL_LITERACY, SKILL_BASIC))
+			to_chat(user, SPAN_WARNING("You can't even read, yet you want to write a whole educational textbook?"))
+			return
+		if(!user.skill_check(SKILL_LITERACY, SKILL_PROF))
+			to_chat(user, SPAN_WARNING("You have no clue as to how to write an entire textbook in a way that is actually useful. Maybe a regular book would be better?"))
+			return
+
+		var/choice = input("What would you like to change?","Textbook editing") as null|anything in list("Title", "Author", skill_option_string)
+		switch(choice)
+
+			if("Title")
+				var/newtitle = reject_bad_text(sanitizeSafe(input("Write a new title:")))
+				if(!newtitle)
+					to_chat(usr, "The title is invalid.")
+					return
+				else
+					newtitle = usr.handle_writing_literacy(usr, newtitle)
+					if(newtitle)
+						title = newtitle
+						SetName(title)
+
+			if("Skill")
+				if(user.skillset.literacy_charges <= 0)
+					to_chat(user, SPAN_WARNING(pick("Your mind instantly recoils at the idea of having to write another entire textbook. No thank you!","You are far too mentally exhausted to write another textbook. Maybe another day.")))
+					return
+
+				//Choosing the skill
+				var/list/skill_choices = list()
+				for(var/decl/hierarchy/skill/S in GLOB.skills)
+					if(user.skill_check(S.type, SKILL_BASIC))
+						skill_choices += S
+				var/decl/hierarchy/skill/skill_choice = input(usr, "What subject does your textbook teach?", "Textbook skill selection") as null|anything in skill_choices
+				if(!skill_choice)
+					to_chat(user, SPAN_WARNING("Textbook skill selection cancelled."))
+					return
+				var/newskill = skill_choice.type
+
+				//Choosing the level
+				var/list/skill_levels = list("Basic" = SKILL_BASIC, "Trained" = SKILL_ADEPT, "Experienced" = SKILL_EXPERT, "Master" = SKILL_PROF)
+				for(var/SL in skill_levels)
+					if(!user.skill_check(newskill, skill_levels[SL]))
+						LAZYREMOVE(skill_levels, SL)
+				var/newskill_level = input(usr, "What level of education does it provide?","Textbook skill level") as null|anything in skill_levels
```

Comment:
You'll probably want/need to check CanPhysicallyInteract and the presence of the pen every time you do `input()`.